projects/storefrontlib/src/cms-structure/services/feature-modules.service.ts

```diff
@@ -0,0 +1,254 @@
+   */
+  getInjectors(componentType: string): Injector[] {
+    const featureName = this.componentFeatureMap.get(componentType);
+    return this.features.get(featureName)?.injectors;
+  }
+
+  private resolveFeature(featureName: string): Observable<FeatureInstance> {
+    return defer(() => {
+      if (this.features.has(featureName)) {
+        return of(this.features.get(featureName));
+      }
+
+      if (!this.featureResolvers.has(featureName)) {
+        const featureConfig = this.featureModulesConfig[featureName];
+
+        if (!featureConfig?.module) {
+          throw 'No module defined for Feature Module ' + featureName;
+        }
+
+        const depsResolve = featureConfig.dependencies?.length
+          ? forkJoin(
+              featureConfig.dependencies.map((depModuleFunc) =>
+                this.resolveDependencyModule(depModuleFunc)
+              )
+            )
+          : of(undefined);
+
+        this.featureResolvers.set(
+          featureName,
+          depsResolve.pipe(
+            switchMap((deps) => this.resolveFeatureModule(featureConfig, deps)),
+            tap((featureInstance) => {
+              this.features.set(featureName, featureInstance);
+
+              this.featureResolvers.delete(featureName);
+            }),
+            share()
+          )
+        );
+      }
+      return this.featureResolvers.get(featureName);
+    });
+  }
+
+  private resolveFeatureModule(
+    featureConfig: FeatureModuleConfig,
+    depsModules: any[] = []
+  ): Observable<FeatureInstance> {
+    return from(featureConfig?.module()).pipe(
+      switchMap((module) =>
+        module instanceof NgModuleFactory
+          ? of(module)
+          : from(this.compiler.compileModuleAsync(module as any))
+      ),
+      observeOn(queueScheduler),
+      map((moduleFactory: NgModuleFactory<any>) => {
+        const moduleRef = moduleFactory.create(this.injector);
+
+        const injectors = [
+          moduleRef.injector,
+          ...depsModules.map(
+            (module) => this.dependencyModules.get(module).injector
+          ),
+        ];
+
+        const featureInstance: FeatureInstance = {
+          ...featureConfig,
+          moduleRef,
+          depsModules,
+          injectors,
+          componentsMappings: {},
+        };
+
+        // resolve configuration for feature module
+        const resolvedConfiguration = this.resolveFeatureConfiguration(
+          moduleRef.injector
+        );
+
+        // extract cms components configuration from feature config
+        for (const componentType of featureInstance.cmsComponents) {
+          featureInstance.componentsMappings[componentType] =
+            resolvedConfiguration.cmsComponents[componentType];
+        }
+        return featureInstance;
+      })
+    );
+  }
+
+  /**
+   * Returns feature configuration by composing root configuration tokens with
+   * configuration tokens retrieved from feature library
```

Comment:
This comment is misleading, becuase the implementation merges only SELF-provided chunks that should be ONLY cms-components mapping related. IMO it's important to note here.

Reply:
:+1: It was outdated indeed. Fixed.